zio-http/src/main/scala/zhttp/service/client/content/handlers/ClientSocketUpgradeHandler.scala

```diff
@@ -0,0 +1,32 @@
+package zhttp.service.client.content.handlers
+
+import io.netty.buffer.Unpooled
+import io.netty.channel.{ChannelHandlerContext, SimpleChannelInboundHandler}
+import io.netty.handler.codec.http.FullHttpResponse
+import zhttp.http.{Headers, Status}
+import zhttp.service.Client.ClientResponse
+import zhttp.service.HttpRuntime
+import zio.Promise
+
+final case class ClientSocketUpgradeHandler[R](
+  zExec: HttpRuntime[R],
+  pr: Promise[Throwable, ClientResponse],
+) extends SimpleChannelInboundHandler[FullHttpResponse] {
+  override def channelRead0(ctx: ChannelHandlerContext, msg: FullHttpResponse): Unit = {
+    msg.touch()
```

Comment:
To add hint information here for debugging purposes, will help in case FullHttpResponse leaks.